src/documentation/markdown.rs

```diff
@@ -382,4 +418,107 @@ And a line, or a rule."##;
+                MARKDOWN[markdown_range.clone()].to_owned()
+            );
+        }
+    }
+
+    #[test]
+    fn markdown_reduction_mapping_auto_link() {
+        const MARKDOWN: &str = r#" <http://foo.bar/baz>"#;
+        const PLAIN: &str = r#""#;
+
+        let (plain, mapping) = PlainOverlay::extract_plain_with_mapping(MARKDOWN);
+
+        assert_eq!(dbg!(&plain).as_str(), PLAIN);
+        assert_eq!(dbg!(&mapping).len(), 0);
+        for (reduced_range, markdown_range) in mapping.iter() {
+            assert_eq!(plain[reduced_range.clone()].to_owned(), String::from(""));
+        }
+    }
+
+    #[test]
+    fn markdown_reduction_mapping_auto_link_with_misspelled_words() {
+        const MARKDOWN: &str = r#" dyrck <http://foo.bar/baz> artic"#;
+        const PLAIN: &str = r#"dyrck  artic"#;
+
+        let (plain, mapping) = PlainOverlay::extract_plain_with_mapping(MARKDOWN);
+
+        assert_eq!(dbg!(&plain).as_str(), PLAIN);
+        assert_eq!(dbg!(&mapping).len(), 2);
+        for (reduced_range, markdown_range) in mapping.iter() {
+            assert_eq!(
+                plain[reduced_range.clone()].to_owned(),
+                MARKDOWN[markdown_range.clone()].to_owned()
+            );
+        }
+    }
+
+    #[test]
+    fn markdown_reduction_mapping_email_with_misspelled_words() {
+        const MARKDOWN: &str = r#" dyrck <joe@example.com> artic"#;
+        const PLAIN: &str = r#"dyrck  artic"#;
+
+        let (plain, mapping) = PlainOverlay::extract_plain_with_mapping(MARKDOWN);
+
+        assert_eq!(dbg!(&plain).as_str(), PLAIN);
+        assert_eq!(dbg!(&mapping).len(), 2);
+        for (reduced_range, markdown_range) in mapping.iter() {
+            assert_eq!(
+                plain[reduced_range.clone()].to_owned(),
+                MARKDOWN[markdown_range.clone()].to_owned()
+            );
+        }
+    }
+
+    #[test]
+    #[ignore]
+    fn markdown_reduction_mapping_footnote() {
+        // What is a markdown footnote syntax? Example:
+        //
+        // Here's a longer one.[^bignote]
+        // [^bignote]: Here's one with multiple paragraphs and code.
+        //
+        // From the pulldown-cmark anything can be followed in the footnote
+        // https://github.com/raphlinus/pulldown-cmark/blob/ad91999dbb0cf0ce64785ed8d22180064985b95a/src/parse.rs#L347
+        //
+        // from this test example, the event has the following form:
+        // Event: FootnoteReference(Borrowed("reference"))
+        // There are two Footnote events as we have [^reference] twice
+        // and the following text is considered as text and not part of the footnote by the parsing
+        //
+        // This followed text has no distinguishing of link or normal text.
+        // Maybe this need more thought about it as the parsing will not tell me if is a link.
+        //
+        // Also, should the reference name be checked or not? If yes, then the mapping needs to be fixed as it have some extras "[]"
```

Comment:
The reference name should be checked, the url checking would be part of #44 .

Reply:
ok it is clear for me!